lib/rogue/models/coordinate.rb

```diff
@@ -5,94 +5,72 @@ def initialize(x, y)
 
   attr_reader :x, :y
 
-  CHASE_DIRECTIONS = {:above? => :down, :below? => :up, :left_of? => :right, :right_of? => :left, \
-  :topleft_of? => :bottomright, :topright_of? => :bottomleft, :bottomright_of? => :topleft, :bottomleft_of? => :topright}
-
-  def eql?(object)
-    object.x == x && object.y == y
-  end
-
-  def hash
-    [x, y].hash
-  end
-
-  def up
-    Coordinate.new(x, y-1)
-  end
-
-  def down
-    Coordinate.new(x, y+1)
-  end
-
-  def left
-    Coordinate.new(x-1, y)
-  end
-
-  def right
-    Coordinate.new(x+1, y)
-  end
-
-  def topleft
-    Coordinate.new(x-1, y-1)
-  end
-
-  def topright
-    Coordinate.new(x+1, y-1)
-  end
-
-  def bottomleft
-    Coordinate.new(x-1, y+1)
-  end
-
-  def bottomright
-    Coordinate.new(x+1, y+1)
+  DIRECTIONS = [:left, :topleft, :up, :topright, :right, :bottomright, :down, :bottomleft]
+  DIRECTIONS_DIAGONALS_FIRST = DIRECTIONS.partition { |dir| DIRECTIONS.index(dir).odd? }.flatten
+
+  ROTATE_PATTERN = [-1, -1, 0, +1, +1, +1, 0, -1]
+  X_ROTATION = ROTATE_PATTERN
+  Y_ROTATION = X_ROTATION.rotate(-2)
+
+  def method_missing(direction, *args)
```

Comment:
hmm, I get this but I think it adds a level of complexity which isn't really beneficial in this case, given that it's a pretty simple class. method_missing also makes it difficult to debug (by messing up the stack trace when an error does occur), I'd go for keeping things explicit even if it's more lines of code.